src/implementation.js

```diff
@@ -1,18 +1,86 @@
 'use strict';
 
-/* Example singleton with es6
- * let myInstance = null;
- * const SomeClass = require('SomeClass');
- *
- * module.exports = function() {
- *     if (!myInstance) {
- *         myInstance = new SomeClass();
- *     }
- *
- *     return myInstance;
- * }
- *
-**/
-
-//export result here
-module.exports = {};
+const Templater = require('toki-templater');
+const Request = require('superagent');
+const Joi = require('joi');
+const Promise = require('bluebird');
+
+const configSchema = Joi.object().keys({
+    name: Joi.string(),
+    url: Joi.string().uri().required(),
+    passThroughHeaders: Joi.alternatives().try(Joi.boolean(), Joi.array()),
+    headers: Joi.object(),
+    payload: Joi.any(),
+    method: Joi.string().allow(['get', 'post', 'put', 'del', 'patch']).default('get'),
+    type: Joi.string().allow(['json', 'application/json']),
+    clientResponse: Joi.alternatives().try(Joi.boolean(), Joi.object()),
+    output: Joi.any()
```

Comment:
Same question here to rename `output` to `assembleConfiguration`. I think the idea was that we can set a standard for method configs so that devs don't have to remember which method calls the input/output mappings by which names.